mne_connectivity/decoding/tests/test_decomposition.py

```diff
@@ -0,0 +1,691 @@
+    n_seeds = 3
+    n_targets = 3
+    fmin = 15
+    fmax = 20
+    epochs = make_signals_in_freq_bands(
+        n_seeds=n_seeds, n_targets=n_targets, freq_band=(fmin, fmax), rng_seed=44
+    )
+    indices = (np.arange(n_seeds), np.arange(n_targets) + n_seeds)
+    cwt_freqs = np.arange(fmin, fmax + 0.5, 0.5)
+    cwt_n_cycles = 6
+
+    # TEST BAD INITIALISATION
+    # Test info
+    with pytest.raises(TypeError, match="`info` must be an instance of mne.Info"):
+        CoherencyDecomposition(info="info", method=method, indices=indices)
+
+    # Test indices
+    with pytest.raises(
+        TypeError, match="`indices` must be an instance of tuple of array-likes"
+    ):
+        CoherencyDecomposition(info=epochs.info, method=method, indices=list(indices))
+    with pytest.raises(
+        TypeError, match="`indices` must be an instance of tuple of array-likes"
+    ):
+        CoherencyDecomposition(info=epochs.info, method=method, indices=(0, 1))
+    with pytest.raises(ValueError, match="`indices` must have length 2"):
+        CoherencyDecomposition(info=epochs.info, method=method, indices=(indices[0],))
+    with pytest.raises(
+        ValueError,
+        match=(
+            "multivariate indices cannot contain repeated channels within a seed or "
+            "target"
+        ),
+    ):
+        CoherencyDecomposition(
+            info=epochs.info, method=method, indices=([0, 0], [1, 2])
+        )
+    with pytest.raises(
+        ValueError,
+        match=(
+            "multivariate indices cannot contain repeated channels within a seed or "
+            "target"
+        ),
+    ):
+        CoherencyDecomposition(
+            info=epochs.info, method=method, indices=([0, 1], [2, 2])
+        )
+    with pytest.raises(
+        ValueError, match="a negative channel index is not present in the data"
+    ):
+        CoherencyDecomposition(
+            info=epochs.info, method=method, indices=([0], [(n_seeds + n_targets) * -1])
+        )
+    with pytest.raises(
+        ValueError,
+        match=(
+            "at least one entry in `indices` is greater than the number of channels in "
+            "`info`"
+        ),
+    ):
+        CoherencyDecomposition(
+            info=epochs.info, method=method, indices=([0], [n_seeds + n_targets])
+        )
+
+    # Test mode
+    with pytest.raises(ValueError, match="Invalid value for the 'mode' parameter"):
+        CoherencyDecomposition(
+            info=epochs.info, method=method, indices=indices, mode="notamode"
+        )
+
+    # Test fmin & fmax
+    if mode in ["multitaper", "fourier"]:
+        with pytest.raises(
+            TypeError,
+            match=(
+                "`fmin` and `fmax` must not be None if `mode` is 'multitaper' or "
+                "'fourier'"
+            ),
+        ):
+            CoherencyDecomposition(
+                info=epochs.info,
+                method=method,
+                indices=indices,
+                mode=mode,
```

Comment:
If almost all of these take these four arguments perhaps it becomes more readable to do:
```
                **base_kwargs
```
and earlier have
```
base_kwargs = dict(
    info=epochs.info,
    method=method,
    indices=indices,
    mode=mode,
)
```
And as a nice side effect hopefully a lot of these calls can become one-liners rather than taking 5+ lines each.

Reply:
Good idea. Addressed in 88eafb5de21d2aba90cd23d7ad913888755a6332 & looks much nicer!